From a74097acb82f156f19b881bdcbab6ad1400d2065 Mon Sep 17 00:00:00 2001
From: Antony Abee Prakash XV <antonyabee.prakashxv@adlinktech.com>
Date: Thu, 6 Aug 2020 18:32:58 +0530
Subject: [PATCH 29/29] Add option to activate alternative 'coda' VPU driver

Freescale defines 2 alternative DT nodes for the VPU: The one normally active
loads a driver called 'mxc_vpu'. The other, disabled one, would invoke a
different driver named 'coda'. None of Freescale's sample dts files enable it,
but the code is there, appearently dormant, in imx6qdl.dtsi.

This patch will add option in device tree to activate alternate 'coda' vpu driver.

Signed-off-by: Antony Abee Prakash XV <antonyabee.prakashxv@adlinktech.com>
---
 arch/arm/boot/dts/lec-imx6.dtsi | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/arch/arm/boot/dts/lec-imx6.dtsi b/arch/arm/boot/dts/lec-imx6.dtsi
index c5103acf37ab..c25a1d484cfd 100644
--- a/arch/arm/boot/dts/lec-imx6.dtsi
+++ b/arch/arm/boot/dts/lec-imx6.dtsi
@@ -11,6 +11,14 @@
 #define CONFIG_SER_CAMERA      3c /* I2C addr of serial camera, hex without "0x" */
 //#define CONFIG_PAR_CAMERA    3c /* I2C addr of parallel camera */
 
+
+/* Freescale defines 2 alternative DT nodes for the VPU: The one normally active
+ * loads a driver called 'mxc_vpu'. The other, disabled one, would invoke a
+ * different driver named 'coda'. None of Freescale's sample dts files enable it, 
+ * but the code is there, appearently dormant, in imx6qdl.dtsi. 
+*/
+#define VPU_CODA       0
+
 #include <dt-bindings/clock/imx6qdl-clock.h>
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/input/input.h>
@@ -321,6 +329,15 @@
 	//status = "okay";
 };
 
+#if VPU_CODA
+&vpu {
+	status = "okay";
+};
+&vpu_fsl {
+	status = "disabled";
+};
+#endif
+
 #if CONFIG_I2S_AUDIO
 &audmux {	/* SMARC I2S0 (instead of SPI0) */
 	pinctrl-names = "default";
-- 
2.17.1

